### PR TITLE
[refactor] Improves icon position calculation

### PR DESCRIPTION
**Pull request by ingoncalves**
This refactoring takes advantage of the new [ep_script_dimensions](https://github.com/storytouch/ep_script_dimensions) plugin, which calculates the user lines within the pad.  With this information we can improve the algorithm that calculates the icon's position, avoiding calling browser functions that cause poor performance (see https://gist.github.com/paulirish/5d52fb081b3570c81e3a).

Depends on:
 - https://github.com/storytouch/ep_script_dimensions/pull/4
 - https://github.com/storytouch/ep_script_dimensions/pull/5

Implements part of the [card 2348](https://trello.com/c/8DybXWnI).

### CODE REVIEW

**Review by WillianGoncalves — APPROVED** [review]
LGTM

File: static/js/commentIcons.js [hunk 1]
```diff
@@ -1,110 +1,95 @@
...
+var FIRST_LINE_OF_PAD = 0;
+var TIME_TO_UPDATE_ICON_POSITION = 100; // milliseconds
 
 var commentIcons = function(ace) {
-  this.thisPlugin = pad.plugins.ep_comments_page;
-  this.commentInfoDialog = this.thisPlugin.commentInfoDialog;
-  this._insertContainer(); // create the container
-
-  // allow this time to be changed if we need to speed things up
+  // allow to controll this timeout in test
```
Comment (joassouza) [inline static/js/commentIcons.js]: ~controll~ control

File: static/js/commentDataManager.js [hunk 2]
```diff
@@ -206,33 +209,48 @@ commentDataManager.prototype._resetRepliesOnComments = function() {
...
   // TODO this method is doing too much. On this case we only need to send the list
   // of comments on the api, don't need to collect all comments from text
-  this.updateListOfCommentsStillOnText();
+  var commentsOnText = textMarkOccurrences ? textMarkOccurrences : this.getTextMarkOccurrencesOnText();
+  this._updateListOfCommentsStillOnText(commentsOnText);
 }
 
 // some comments might had been removed from text, so update the list
-commentDataManager.prototype.updateListOfCommentsStillOnText = function() {
+commentDataManager.prototype._updateListOfCommentsStillOnText = function(textMarkOccurrences) {
```
Comment (joassouza) [inline static/js/commentDataManager.js:223]: tô com uma dúvida nessa função aqui. Não sei se o nome faz sentido mais...
A gente constrói a lista aqui  no módulo `textMarksFinder`

Comment (ingoncalves) [reply static/js/commentDataManager.js:223]: Adicionei um comentário explicando.

**Pull request merged**